### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by kene-creator**
Carried out a unit test for all my functions including:
- books
- rental
- decorators
- teacher
- student

### CODE REVIEW

**Comment by DammyShittu** [comment]
Hi @kene-creator,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### You need to build this project with your coding partner

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      expect(person.can_use_service?).to eq(true)
+    end
+
+    it 'returns false if the person is not of age and has no parent permission' do
+      person.age = 17
+      person.instance_variable_set(:@parent_permission, false)
+      expect(person.can_use_service?).to eq(false)
+    end
+  end
+end
```
Comment (TedLivist) [inline spec/person_spec.rb]: - [x] All public methods should be tested. Kindly add tests for the `add_rental` in your Person class.

File: spec/book_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      expect(book.title).to eq('The Great Gatsby')
+      expect(book.author).to eq('F. Scott Fitzgerald')
+    end
+
+    it 'initializes the rentals attribute as an empty array' do
+      book = Book.new(title: 'The Great Gatsby', author: 'F. Scott Fitzgerald')
+      expect(book.rentals).to eq([])
+    end
+  end
+end
```
Comment (TedLivist) [inline spec/book_spec.rb]: - [ ] All public methods should be tested. Kindly add tests for the `add_rental` in your Book class.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @kene-creator and @Fayob,

Good job so far!
✔️ Good documentation
✔️ Use rspec for testing
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [x] Kindly add tests for the classroom class.

- [x] Some of your tests are falling. Kindly fix them.

  suwoe

- [x] _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @kene-creator and @Fayob,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/book_spec.rb (hunk 2 above)
Comment (kene-creator) [reply spec/book_spec.rb]: Thank you. I mistakenly missed out on that one.

**Pull request merged**